redefinition of 'math__bits__mul_64'

### Describe the bug

When create a file, `math/bits/bits.amd.v` and place some funcs in it, it will cause v compile error when work with `-translated`.
It seems related to `const` declare in the test file.

### Reproduction Steps

create a  file `vlib/math/bits/bits.amd.v`:

```v
module bits

pub fn mul_64(x u64, y u64) (u64, u64) {
        // fallback to default
        x0 := x & mask32
        x1 := x >> 32
        y0 := y & mask32
        y1 := y >> 32
        w0 := x0 * y0
        t := x1 * y0 + (w0 >> 32)
        mut w1 := t & mask32
        w2 := t >> 32
        w1 += x0 * y1
        hi := x1 * y1 + w2 + (w1 >> 32)
        lo := x * y
        return hi, lo
}
```

create a test file, `t.v`:
```v
module main

const ssf = [1, 2, 3]

fn main() {
        assert ssf[0] == 1
        println('ok')
}
```

compile it :
```
v -translated t.v
```

### Expected Behavior

compile OK

### Current Behavior

```
$ v -translated t.v
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/t.01K1M4N9HASBF4P1ZYGDND42AR.tmp.c:1476: error: redefinition of 'math__bits__mul_64'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .

```

### Possible Solution

_No response_

### Additional Information/Context

1. Just compile without `-translated` is OK;
2. Comment out the `const`, it will compile OK.


### V version

V 0.4.11 019fc8e7c858afa967c3ada4503c3af7cc5d4bf1.6c72feb

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.11 019fc8e7c858afa967c3ada4503c3af7cc5d4bf1.6c72feb
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.2 LTS
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
|Memory              |8.79GB/15.51GB
|                    |
|V executable        |/media/HD/github/kbkpbot/v/v
|V last modified time|2025-08-02 00:40:51
|                    |
|V home dir          |OK, value: /media/HD/github/kbkpbot/v
|VMODULES            |OK, value: /home/mars/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/mars/v/bug/translated
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2025.17-490-g6c72febe-dirty
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |Ubuntu clang version 18.1.3 (1ubuntu1)
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.5) 2.39

--- a/vlib/v/parser/fn.v
+++ b/vlib/v/parser/fn.v
@@ -584,7 +584,7 @@ run them via `v file.v` instead',
 			.wasm { 'WASM.${name}' }
 			else { p.prepend_mod(name) }
 		}
-		if !p.pref.translated && language == .v {
+		if language == .v {
 			if existing := p.table.fns[name] {
 				if existing.name != '' {
 					if file_mode == .v && existing.file_mode != .v {

--- a/vlib/v/parser/fn.v
+++ b/vlib/v/parser/fn.v
@@ -592,7 +592,7 @@ run them via `v file.v` instead',
 						if !p.pref.is_fmt {
 							name = p.prepend_mod('pure_v_but_overridden_by_${existing.file_mode}_${short_fn_name}')
 						}
-					} else {
+					} else if !p.pref.translated {
 						p.table.redefined_fns << name
 					}
 				}

